src/shared/hooks/useAsync/useAsync.ts

```diff
@@ -1,38 +1,51 @@
-import { useCallback, useRef, useState } from 'react';
+import { useCallback, useState } from 'react';
 import { AxiosError, AxiosResponse } from 'axios';
 
 import { ApiErrorResponse } from 'shared/state/Base';
 
+import {
+  ErrorCallback,
+  FinallyCallback,
+  optionsIsObjectTypeGuard,
+  SuccessCallback,
+  UseAsyncOptions,
+} from './useAsync.types';
+
 export const useAsync = <T, K>(
```

Comment:
`useAsync` is a highly used hook used in almost 100 places. Might be a good opportunity to include a couple of lines of documentation. The options are straightforward except for `retainValue` which
```suggestion
/**
 * Custom hook to handle asynchronous operations with callbacks,
 * and a retainValue flag to keep the previous value during the loading state.
 */
export const useAsync = <T, K>(
```